statsmodels/discrete/truncated_model.py

```diff
@@ -0,0 +1,1412 @@
+
+        if dist == "poisson":
+            self.model2 = TruncatedPoisson(self.endog, self.exog)
+        elif dist == "negbin":
+            self.model2 = TruncatedNegativeBinomialP(self.endog, self.exog,
+                                                     p=p)
+
+    def loglike(self, params):
+        """
+        Loglikelihood of Generic Hurdle model
+
+        Parameters
+        ----------
+        params : array-like
+            The parameters of the model.
+
+        Returns
+        -------
+        loglike : float
+            The log-likelihood function of the model evaluated at `params`.
+            See notes.
+
+        Notes
+        --------
+
+        """
+        k = int((len(params) - self.k_extra1 - self.k_extra2) / 2
+                ) + self.k_extra1
+        return (self.model1.loglike(params[:k]) +
+                self.model2.loglike(params[k:]))
+
+    def fit(self, start_params=None, method='bfgs', maxiter=35,
+            full_output=1, disp=1, callback=None,
+            cov_type='nonrobust', cov_kwds=None, use_t=None, **kwargs):
+
+        results1 = self.model1.fit(
+            start_params=start_params,
+            method=method, maxiter=maxiter, disp=disp,
+            full_output=full_output, callback=lambda x: x,
+            **kwargs
+            )
+
+        results2 = self.model2.fit(
+            start_params=start_params,
+            method=method, maxiter=maxiter, disp=disp,
+            full_output=full_output, callback=lambda x: x,
+            **kwargs
+            )
+
+        result = deepcopy(results1)
+        result._results.model = self
+        result.mle_retvals['converged'] = [results1.mle_retvals['converged'],
+                                           results2.mle_retvals['converged']]
+        result._results.params = np.append(results1._results.params,
+                                           results2._results.params)
+        # TODO: the following should be in __init__ or initialize
+        result._results.df_model += results2._results.df_model
+        self.k_extra1 += getattr(results1._results, "k_extra", 0)
+        self.k_extra2 += getattr(results2._results, "k_extra", 0)
+        self.k_extra = (self.k_extra1 + self.k_extra2 + 1)
+
+        # fix up cov_params
+        from scipy.linalg import block_diag
+        result._results.normalized_cov_params = (
+            block_diag(results1._results.cov_params(),
+                       results2._results.cov_params()))
+
+        modelfit = self.result(self, result._results, results1, results2)
+        result = self.result_wrapper(modelfit)
+
+        if cov_kwds is None:
+            cov_kwds = {}
+
+        result._get_robustcov_results(cov_type=cov_type,
+                                      use_self=True, use_t=use_t, **cov_kwds)
+        return result
+
+    fit.__doc__ = DiscreteModel.fit.__doc__
+
+    def predict(self, params, exog=None, exog_zero=None, exposure=None,
```

Comment:
I started to add exog_zero here (in analogy to ZI), but it's not in the `__init__` yet